packages/flutter/lib/src/widgets/image.dart

```diff
@@ -108,6 +108,88 @@ Future<void> precacheImage(
+/// no [Image.frameBuilder] had been specified. In other words, returning
+/// `child` directly from this method will yield the same result as this method
+/// having not been consulted. This argument is guaranteed to be non-null.
+/// Typically, this builder will wrap the `child` widget in some way and return
+/// the wrapped widget.
+///
+/// The `frame` argument specifies the index of the current image frame being
+/// rendered. It will be null before the first image frame is ready, and zero
+/// for the first image frame. For single-frame images, it will never be greater
+/// than zero. For multi-frame images (such as animated GIFs), it will increase
+/// by one every time a new image frame is shown (including when the image
+/// animates in a loop).
+///
+/// The `wasSynchronouslyLoaded` argument specifies whether the image was
+/// available synchronously (on the same
+/// [rendering pipeline frame](rendering/RendererBinding/drawFrame.html) as the
+/// `Image` widget itself was created) and thus able to be painted immediately.
+/// If this is false, then there was one or more rendering pipeline frames where
+/// the image wasn't yet available to be painted. For multi-frame images (such
+/// as animated GIFs), the value of this argument will be the same for all image
+/// frames. In other words, if the first image frame was available immediately,
+/// then this argument will be true for all image frames.
+///
+/// See also:
+///
+///  * [Image.frameBuilder], which makes use of this signature in the [Image]
+///    widget.
+typedef ImageFrameBuilder = Widget Function(
+  BuildContext context,
+  Widget child,
+  int frame,
+  bool wasSynchronouslyLoaded,
+);
+
+/// Signature used by [Image.loadingBuilder] to build a representation of the
+/// image's loading progress.
+///
+/// This is useful for images that are incrementally loaded (e.g. over a local
+/// file system or a network), and the application wishes to give the user an
+/// indication of when the image will be displayed.
+///
+/// The `child` argument contains the default widget that would be rendered if
+/// no [Image.loadingBuilder] had been specified. In other words, returning
+/// `child` directly from this method will yield the same result as this method
+/// having not been consulted. This argument is guaranteed to be non-null.
+/// Typically, this builder will wrap the `child` widget in some way and return
+/// the wrapped widget.
+///
+/// The `loadingProgress` argument contains the current progress towards loading
+/// the image. This argument will be non-null while the image is loading, but it
+/// will be null in the following cases:
+///
+///   * When the widget is first rendered before any bytes have been loaded.
+///   * When an image frame has been fully loaded and is available to be painted,
+///     and before any further [ImageChunkEvent]s have been received.
+///
+/// It follows that for single-frame images, or multi-frame images whose bytes
```

Comment:
..whose bytes for *all* frames... (not just the first, right)?

Reply:
Yep.  Clarified.